public/app/features/alerting/unified/components/rule-list/RuleList.v2.tsx

```diff
@@ -0,0 +1,219 @@
+
+    const allRulerEmpty = rulerRequests.every(([_, state]) => {
+      const rulerRules = Object.entries(state?.result ?? {});
+      const noRules = rulerRules.every(([_, result]) => result?.length === 0);
+      return noRules && state.dispatched;
+    });
+
+    const limitAlerts = hasActiveFilters ? undefined : LIMIT_ALERTS;
+    // Trigger data refresh only when the RULE_LIST_POLL_INTERVAL_MS elapsed since the previous load FINISHED
+    const [_, fetchRules] = useAsyncFn(async () => {
+      if (!loading) {
+        await dispatch(fetchAllPromAndRulerRulesAction(false, { limitAlerts }));
+      }
+    }, [loading, limitAlerts, dispatch]);
+
+    useEffect(() => {
+      trackRuleListNavigation().catch(() => {});
+    }, []);
+
+    // fetch rules, then poll every RULE_LIST_POLL_INTERVAL_MS
+    useEffect(() => {
+      dispatch(fetchAllPromAndRulerRulesAction(false, { limitAlerts }));
+    }, [dispatch, limitAlerts]);
+    useInterval(fetchRules, RULE_LIST_POLL_INTERVAL_MS);
+
+    // Show splash only when we loaded all of the data sources and none of them has alerts
+    const hasNoAlertRulesCreatedYet =
+      allPromLoaded && allPromEmpty && promRequests.length > 0 && allRulerEmpty && allRulerLoaded;
+    const hasAlertRulesCreated = !hasNoAlertRulesCreatedYet;
+
+    const combinedNamespaces: CombinedRuleNamespace[] = useCombinedRuleNamespaces();
+    const filteredNamespaces = useFilteredRules(combinedNamespaces, filterState);
+
+    const sortedNamespaces = filteredNamespaces.sort((a: CombinedRuleNamespace, b: CombinedRuleNamespace) =>
+      a.name.localeCompare(b.name)
+    );
+
+    return (
+      // We don't want to show the Loading... indicator for the whole page.
+      // We show separate indicators for Grafana-managed and Cloud rules
+      <AlertingPageWrapper navId="alert-list" isLoading={false} actions={hasAlertRulesCreated && <CreateAlertButton />}>
+        <RuleListErrors />
+        <RulesFilter onFilterCleared={onFilterCleared} />
+        {hasAlertRulesCreated && (
+          <>
+            <div className={styles.break} />
+            <div className={styles.buttonsContainer}>
+              <div className={styles.statsContainer}>
+                {hasActiveFilters && (
+                  <Button
+                    className={styles.expandAllButton}
+                    icon={expandAll ? 'angle-double-up' : 'angle-double-down'}
+                    variant="secondary"
+                    onClick={() => setExpandAll(!expandAll)}
+                  >
+                    {expandAll ? 'Collapse all' : 'Expand all'}
+                  </Button>
+                )}
+                <RuleStats namespaces={filteredNamespaces} />
+              </div>
+            </div>
+          </>
+        )}
+        {hasNoAlertRulesCreatedYet && <NoRulesSplash />}
+        {hasAlertRulesCreated && (
+          <>
+            <LoadingIndicator visible={loading} />
+            <ul className={styles.rulesTree} role="tree">
```

Comment:
Consider adding an `aria-label` or `aria-labeledby` here. The screen reader only announces this as `table`. If possible, maybe add some kind of landmark markup somewhere here too? The tree is the main attraction of the page, but when looking at form elements in the screen reader, it appears quite late.